packages/protocol/lib/web3-utils.ts

```diff
@@ -236,116 +210,33 @@ export async function setInRegistryViaMultiSig(
  * A function with a signature as expected to be exported from a Truffle
  * migration script.
  */
-export function deployProxyAndImplementation<ContractInstance extends Truffle.ContractInstance>(
-  web3: Web3,
-  artifacts: any,
-  name: string,
-  args: (networkName?: string) => Promise<any[]>,
-  then?: (contract: ContractInstance, web3: Web3, networkName: string) => void
-) {
-  return async (deployer: any, networkName: string, _accounts: string[]) => {
-    console.log('Deploying', name)
-    let Contract = artifacts.require(name)
-    let ContractProxy = artifacts.require(name + 'Proxy')
-    try {
-      Contract = artifacts.require(name)
-      ContractProxy = artifacts.require(name + 'Proxy')
-    } catch (e) {
-      console.error(e)
-      process.exit(1)
-    }
-    let proxiedContract: ContractInstance
-    deployer
-      .deploy(ContractProxy)
-      .then(async () => {
-        const proxy: ProxyInstance = await ContractProxy.deployed()
-        await proxy._transferOwnership(_accounts[0])
-      })
-      .then(() => {
-        return deployer.deploy(Contract)
-      })
-      .then(async () => {
-        proxiedContract = await setInitialProxyImplementation<ContractInstance>(
-          web3,
-          artifacts,
-          name,
-          ...(await args(networkName))
-        )
-      })
-      .then(async () => {
-        if (then) {
-          return then(proxiedContract, web3, networkName)
-        }
-      })
-  }
-}
-
-/*
- * Used to deploy contracts whose Proxies have already been deployed.
- *
- * Arguments:
- * - artifacts: the Resolver object provided by Truffle
- * - name: name of the contract to deploy
- * - args: array of arguments to the contract's initializer
- * - then: a callback that can perform additional migration operations after deployment
- *
- * The callback will be called with the deployed proxied contract, a web3
- * instance (derived from the provider of the deployer given by Truffle), and the
- * name of the network (as given by Truffle).
- *
- * Returns:
- * A function with a signature as expected to be exported from a Truffle
- * migration script.
- */
-export function deployImplementationAndRepointProxy<
-  ContractInstance extends Truffle.ContractInstance
->(
+export function deployerForCoreContract<ContractInstance extends Truffle.ContractInstance>(
```

Comment:
What i wanted to express is:

 - it's not deploying, it just return a deploy function (that's why "deployed" and not "deploy")
 - Instead of listing the steps (1. deploy impl, 2. deploy proxy, 3. set impl 4 (new one). set registry. I wanted to name the process "deployCoreContract".. whatever that means

Reply:
changed to `deploymentFor`